x-pack/platform/plugins/shared/fields_metadata/server/services/fields_metadata/repositories/otel_fields_repository.ts

```diff
@@ -87,7 +87,9 @@ export class OtelFieldsRepository {
       const field = this.getByName(fieldName);
 
       if (field) {
-        fieldsMetadata[fieldName] = field;
+        // Use the stripped field name as the key for consistency
+        const strippedFieldName = stripOtelPrefixes(fieldName as string);
```

Comment:
Not sure I would do this, it would also force the client to strip the prefixes to access the right property on the response, and make the request/response inconsistent... if the consumer of this service requests the `resource.attributes.cloud.account.id` field, I think it's correct to send back the result for what he requested, the fact we resolve aliases and strip prefixes should be transparent for the client.